exchanges/okcoin/okcoin_websocket.go

```diff
@@ -295,27 +264,104 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
 	return nil
 }
 
-func (o *OKCoin) wsProcessAdvancedAlgoOrder(respRaw []byte) error {
+func (o *Okcoin) wsProcessAdvancedAlgoOrder(respRaw []byte) error {
 	var resp WebsocketAdvancedAlgoOrder
 	err := json.Unmarshal(respRaw, &resp)
 	if err != nil {
 		return err
 	}
-	o.Websocket.DataHandler <- resp
+	algoOrders := make([]order.Detail, len(resp.Data))
+	for d := range resp.Data {
+		oType, err := order.StringToOrderType(resp.Data[d].OrderType)
+		if err != nil {
+			return err
+		}
+		oSide, err := order.StringToOrderSide(resp.Data[d].Side)
+		if err != nil {
+			return err
+		}
+		// this code block introduces two order states
+		// 1. 'effective' - equivalent to filled, and
+		// 2. 'order_failed' == equivalent to failed
+		oStatus, err := order.StringToOrderStatus(resp.Data[d].State)
+		if err != nil {
+			return err
+		}
+		cp, err := currency.NewPairFromString(resp.Data[d].InstrumentID)
+		if err != nil {
+			return err
+		}
+		algoOrders[d] = order.Detail{
+			Leverage:        resp.Data[d].Lever.Float64(),
+			Price:           resp.Data[d].OrderPrice.Float64(),
+			Amount:          resp.Data[d].Size.Float64(),
+			LimitPriceUpper: resp.Data[d].PriceLimit.Float64(),
+			TriggerPrice:    resp.Data[d].TriggerPrice.Float64(),
+			RemainingAmount: resp.Data[d].ActualSz.Float64(),
+			Exchange:        o.Name,
+			OrderID:         resp.Data[d].AlgoID,
+			ClientOrderID:   resp.Data[d].ClOrdID,
+			Type:            oType,
+			Side:            oSide,
+			Status:          oStatus,
+			AssetType:       asset.Spot,
+			Date:            resp.Data[d].CreationTime.Time(),
+			LastUpdated:     resp.Data[d].PushTime.Time().UTC(),
+			Pair:            cp,
+		}
+		o.Websocket.DataHandler <- algoOrders
```

Comment:
This is pushing incomplete data, can you please shift this out of the for loop scope.

Reply:
😲. Alright, sir.